### ISSUE TITLE
dirname() Flips slashes 

### ISSUE THREAD

**Issue by CR-OmerYampel** [Velocidex/velociraptor#960]
https://github.com/Velocidex/velociraptor/blob/f6b51a58dda488d40cd30394fab4f66a528ee44b/vql/functions/paths.go#L48

When using dirname, it flips the slashes from the MFT's output, causing an error when looking for a specific path. For example:

```
SELECT dirname(str=FullPath) as folder FROM parse_mft() WHERE folder = "C:\windows\temp"
```

will not work due to the join being using a forward slash `(components[:len(components)-1], "/")`

I think this should be standardized across all functions (either only use backslashes, or only use forward slashes)

**Comment by scudette** [comment]
Velociraptor does standardize on forward slash, but will also accept backslash. 

You should be able to access the file using either so it doesn't make a difference if it flips the slash does it?

**Comment by CR-OmerYampel** [comment]
The problem isn't the accessing, the problem is the filtering:

```
"c:\windows\system32" == "c:/windows/system32"
```

is false6

**Comment by scudette** [comment]
Ah I see. I guess we can expose the sep as a parameter in the dirname function, but this will only work on future versions. For now simply compare two dirnames together:

```sql
LET subdirectory <= dirname(path='''C:\windows\temp\XX''')
SELECT dirname(path=FullPath) as folder FROM parse_mft() WHERE folder = subdirectory
```

**Comment by scudette** [comment]
Fixed in 0.5.7
